FogCreek/Glitch-Community
# Melted scorpion: fix pagination bug

## Links
* Remix link: https://melted-scorpion.glitch.me/
* Manuscript link: https://glitch.manuscript.com/f/cases/3328769/pagination-bug

## GIF/Screenshots:
This gif is tinier than I thought but if you click it should be more understandable
![](http://g.recordit.co/i27og0JEgq.gif)

## Changes:
* resets the current page to 1 if the number of projects has changed

## How To Test:
* navigate to a user (or team) page or anywhere else we might show paginated and filtered projects
* click forward through a few pages
* find something to filter by that has fewer than that amount of pages
* notice that the the pagination starts at the beginning again (rather than keeping you on a non existent page)

## Feedback I'm looking for:
- I believe I'm using useEffect/useReducer correctly but feel free to tell me if I'm missing something! 
- let me know if you see other bugs this causes! I couldn't find any but it's def possible they are there!
